homeassistant/components/screenlogic/climate.py

```diff
@@ -0,0 +1,214 @@
+        if self.body["heat_status"]["value"] > 0:
+            return CURRENT_HVAC_HEAT
+        if self.hvac_mode == HVAC_MODE_HEAT:
+            return CURRENT_HVAC_IDLE
+        return CURRENT_HVAC_OFF
+
+    @property
+    def preset_mode(self) -> str:
+        """Return current/last preset mode."""
+        if self.hvac_mode == HVAC_MODE_OFF:
+            return HEAT_MODE.NAME_FOR_NUM[self._last_preset]
+        return HEAT_MODE.NAME_FOR_NUM[self.body["heat_mode"]["value"]]
+
+    @property
+    def preset_modes(self):
+        """All available presets."""
+        return [
+            HEAT_MODE.NAME_FOR_NUM[mode_num] for mode_num in self._configured_heat_modes
+        ]
+
+    @property
+    def supported_features(self):
+        """Supported features of the heater."""
+        return SUPPORTED_FEATURES
+
+    async def async_set_temperature(self, temperature, **kwargs) -> None:
+        """Change the setpoint of the heater."""
+        if await self.hass.async_add_executor_job(
+            self.gateway.set_heat_temp, int(self._data_key), int(temperature)
+        ):
+            await self.coordinator.async_request_refresh()
+        else:
+            _LOGGER.warning(
+                "Failed to set_temperature %s on %s",
+                temperature,
+                self.body["body_type"]["value"],
+            )
+
+    async def async_set_hvac_mode(self, hvac_mode) -> None:
+        """Set the operation mode."""
+        if hvac_mode == HVAC_MODE_OFF:
+            mode = HEAT_MODE.OFF
+        else:
+            mode = HEAT_MODE.NUM_FOR_NAME[self.preset_mode]
+        if await self.hass.async_add_executor_job(
+            self.gateway.set_heat_mode, int(self._data_key), int(mode)
+        ):
+            await self.coordinator.async_request_refresh()
+        else:
+            _LOGGER.warning(
+                "Failed to set_hvac_mode %s on %s",
+                mode,
+                self.body["body_type"]["value"],
+            )
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set the preset mode."""
+        _LOGGER.debug("Setting last_preset to %s", HEAT_MODE.NUM_FOR_NAME[preset_mode])
+        self._last_preset = mode = HEAT_MODE.NUM_FOR_NAME[preset_mode]
+        if self.hvac_mode == HVAC_MODE_OFF:
+            return
+        if await self.hass.async_add_executor_job(
+            self.gateway.set_heat_mode, int(self._data_key), int(mode)
+        ):
+            await self.coordinator.async_request_refresh()
+        else:
+            _LOGGER.warning(
+                "Failed to set_preset_mode %s on %s",
+                mode,
+                self.body["body_type"]["value"],
+            )
+
+    async def async_added_to_hass(self):
+        """Run when entity is about to be added."""
+        await super().async_added_to_hass()
+        _LOGGER.debug("Startup last preset is %s", self._last_preset)
+        if self._last_preset is None:
```

Comment:
```suggestion
        if self._last_preset is not None:
            return
```

Then outdent the block below

Reply:
Commit https://github.com/home-assistant/core/pull/48136/commits/2d488e2a1747d24006a99a3b5f374c69f73748c7